cms/djangoapps/contentstore/management/commands/delete_course.py

```diff
@@ -8,27 +15,65 @@
 from opaque_keys import InvalidKeyError
 from opaque_keys.edx.locations import SlashSeparatedCourseKey
 from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+
+
+def _get_all_courses():
+    """
+    Utility function to list all available courses.
+    """
+
+    courses = modulestore().get_courses_keys()
+    return courses
 
 
 class Command(BaseCommand):
+    """
+    Delete a MongoDB backed course
+    """
     help = '''Delete a MongoDB backed course'''
 
     def handle(self, *args, **options):
-        if len(args) != 1 and len(args) != 2:
-            raise CommandError("delete_course requires one or more arguments: <course_id> |commit|")
+        if len(args) == 0:
+            raise CommandError("Arguments missing: 'org/number/run commit'")
 
-        try:
-            course_key = CourseKey.from_string(args[0])
-        except InvalidKeyError:
-            course_key = SlashSeparatedCourseKey.from_deprecated_string(args[0])
+        if len(args) == 1:
+            if args[0] == 'commit':
+                raise CommandError("Delete_courses requires a <course_id> argument. [org/number/run]")
+            else:
+                raise CommandError("Delete_courses requires a commit argument at the end")
+        elif len(args) == 2:
+            try:
+                course_key = CourseKey.from_string(args[0])
+            except InvalidKeyError:
+                try:
+                    course_key = SlashSeparatedCourseKey.from_deprecated_string(args[0])
+                except InvalidKeyError:
+                    raise CommandError("Invalid course_key: '%s'. Proper syntax: 'org/number/run commit' " % args[0])
+            if args[1] != 'commit':
+                raise CommandError("Delete courses requires a commit argument at the end")
 
         commit = False
```

Comment:
Yes, good catch. Also, another case crossed my mind. What if there are more than 2 parameters provided (let's assume that the first two are correct - course ID and "commit")?